src/community/wps-jdbc/src/main/java/org/geoserver/wps/jdbc/JDBCStatusStore.java

```diff
@@ -0,0 +1,493 @@
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } finally {
+            transaction.close();
+        }
+
+    }
+
+    @Override
+    public ExecutionStatus get(String executionId) {
+        LOGGER.info("getting status " + executionId);
+        try {
+            SimpleFeatureSource source = statuses.getFeatureSource(STATUS);
+
+            Filter filter = ECQL.toFilter(PROCESS_ID + " = '" + executionId + "'");
+            SimpleFeatureCollection features = source.getFeatures(filter);
+            SimpleFeature f = DataUtilities.first(features);
+            ExecutionStatus stat = featureToStatus(f);
+            return stat;
+        } catch (IOException | CQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public ExecutionStatus remove(String executionId) {
+        LOGGER.info("removing status " + executionId);
+        try {
+            SimpleFeatureSource source = statuses.getFeatureSource(STATUS);
+
+            Filter filter = ECQL.toFilter(PROCESS_ID + " = '" + executionId + "'");
+            SimpleFeatureCollection features = source.getFeatures(filter);
+            SimpleFeature f = DataUtilities.first(features);
+            ExecutionStatus stat = featureToStatus(f);
+            DefaultTransaction transaction = new DefaultTransaction("create");
+            try {
+
+                if (source instanceof SimpleFeatureStore) {
+                    SimpleFeatureStore store = (SimpleFeatureStore) source;
+                    store.setTransaction(transaction);
+                    store.removeFeatures(filter);
+                    transaction.commit();
+                }
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+                try {
+                    transaction.rollback();
+                } catch (IOException e1) {
+                    // don't care what happens here
+                }
+            } finally {
+                transaction.close();
+            }
+            return stat;
+        } catch (IOException | CQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public int remove(Filter filter) {
+        LOGGER.info("removing statuses matching " + filter);
+        int ret = 0;
+        DefaultTransaction transaction = new DefaultTransaction("create");
+        try {
+            SimpleFeatureSource source = statuses.getFeatureSource(STATUS);
+            SimpleFeatureCollection features = source.getFeatures(filter);
+            ret = features.size();
+            if (ret == 0) {
+                return ret;
+            }
+            SimpleFeatureIterator itr = features.features();
+
+            itr.close();
+            if (source instanceof SimpleFeatureStore) {
```

Comment:
It does do a count and then remove by filter. But there is a spurious iterator left over which I'll remove